src/python/pants/backend/python/providers/python_build_standalone/rules.py

```diff
@@ -178,7 +227,23 @@ def _choose_python(
             )
         )
 
-    return ".".join(map(str, version_triplet))
+    # Choose the highest supported patchlevel of the lowest supported major/minor version
+    # by searching until the major/minor version increases or the search ends (in which case the
+    # last candidate is the one).
+    candidate_pbs_releases.sort(key=lambda x: x[0])
+    major_minor = candidate_pbs_releases[0][0][0:2]
+    i = 0
+    while i < len(candidate_pbs_releases):
+        if (
+            i + 1 < len(candidate_pbs_releases)
+            and candidate_pbs_releases[i + 1][0][0:2] != major_minor
+        ):
+            r = candidate_pbs_releases[i]
+            return (".".join(map(str, r[0])), r[1])
+        i += 1
```

Comment:
From a "Pythonic" standpoint, I think this could be written as 

```suggestion
    for i, (triplet, metadata) in enumerate(candidate_pbs_releases):
        if (
            i + 1 < len(candidate_pbs_releases)
            and candidate_pbs_releases[i + 1][0][0:2] != major_minor
        ):
            return (".".join(map(str, triplet)), metadata)
```

(I think that's the same logic, but not 100% sure.)

We could even "code-golf" this code to something like (this moves the `if not candidate_pbs_release` to after the loop and eliminates the `major_minor` variable too)

```suggestion
    for i, (triplet, metadata) in enumerate(candidate_pbs_releases):
        if (
            # last candidate, we're good!
            i == len(candidate_pbs_releases) - 1
            # next candidate is the next major/minor version, so this is the highest patch
            or candidate_pbs_releases[i + 1][0][0:2] != triplet[0:2]
        ):
            return (".".join(map(str, triplet)), metadata)
            
    # There's no candidates!
    raise Exception(softwrap(f"Failed to find a supported Python Build Standalone ..."))
```

Reply:
Implemented it. Made the final line raise an `AssertionError` since it should be unreachable technically.